src/monitors/Monitor.v3

```diff
@@ -40,3 +40,23 @@ class Monitor {
 	}
 }
 
+component MonitorUtil {
+	def probeOnEnterFunc(module: Module, func: FuncDecl, probe: FuncEnterProbe) {
+		var it = BytecodeIterator.new();
+		var bi = it.reset(func);
+		bi.current();
+		module.insertProbeAt(func.func_index, bi.pc, probe);
+	}
+	def probeOnExitFunc(module: Module, func: FuncDecl, probe: Probe) {
+		var it = BytecodeIterator.new();
+		for (bi = it.reset(func); bi.more(); bi.next()) {
+			var op = bi.current();
+			match (op) {
```

Comment:
One oversight with the existing code I noticed is that branches to block `0` (i.e. the function scope) are not instrumented. I think this is why the calls monitor has been getting misaligned.

Reply:
Does Wizard have a nice way of getting the block depth in `BytecodeIterator`, since the branch label is relative? Or I could write some kind of `BytecodeVisitor` that keeps track of the depth.

This is getting pretty complicated though, especially having to get the values from the opstack for `br_if` and `br_table`s. Not sure if there's a better design you might have in mind.

Reply:
It doesn't yet have such a utility; that would be pretty rad. Maybe a new class `BlockIterator`?

Reply:
Makes sense. Might be useful in `LoopMonitor` too, where I was manually pushing and popping from a control stack to detect nested loops.

Reply:
I'm not sure if this is consistent across execution modes, but it appears that branches to block `0` are already automatically instrumented because we added a probe to the final `end` bytecode. I think it's double-counting exits if I instrument both. Do branches to block `0` hit the final `end` based on your sidetable implementation or am I misunderstanding?

Reply:
Ah yes, you are right. The intention was that they would in fact hit the final `END` bytecode. Probably worth adding an explicit test for this behavior.